packages/eslint-plugin-gestalt/src/prefer-box-as-tag.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview Prefer Box: prevent HTML tags supported in Box through the `as` prop
+ */
+
+// @flow strict
+import { getHtmlTag, hasImport, buildProps } from './eslintASTHelpers.js';
+import { renameTagWithPropsFixer, updateGestaltImportFixer } from './eslintASTFixers.js';
+import { type ESLintRule } from './eslintFlowTypes.js';
+
+export const SUPPORTED_HTML_TAGS = [
+  'article',
+  'aside',
+  'details',
+  'figcaption',
+  'figure',
+  'footer',
+  'header',
+  'main',
+  'nav',
+  'section',
+  'summary',
+];
+
+const rule: ESLintRule = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Prefer Box: prevent HTML tags supported in Box through the `as` prop: "article" , "aside" | "details" | "div" | "figcaption" | "figure" | "footer" | "header" | "main" | "nav" | "section" | "summary", instead',
```

Comment:
nit: rather than repeat the tags here, use a template literal and map the array above to the format needed here

Reply:
good catch!